src/modules/package.rs

```diff
@@ -0,0 +1,89 @@
+use super::Segment;
+use crate::context::Context;
+use ansi_term::Color;
+use serde_json;
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+use toml;
+
+/// Creates a segment with the current package version
+pub fn segment(context: &Context) -> Option<Segment> {
+    match get_package_version(context) {
+        Some(package_version) => {
+            const RUST_CHAR: &str = "📦";
+            const SEGMENT_COLOR: Color = Color::Red;
+
+            let mut segment = Segment::new("package");
+            // TODO: Should have prefix "is "
+            segment.set_style(SEGMENT_COLOR);
+
+            segment.set_value(format!("{} {}", RUST_CHAR, package_version));
+
+            Some(segment)
+        }
+        None => None,
+    }
+}
+
+fn has_rs_files(dir_entry: &PathBuf) -> bool {
+    let is_rs_file =
+        |d: &PathBuf| -> bool { d.is_file() && d.extension().unwrap_or_default() == "rs" };
+    let is_cargo_toml =
+        |d: &PathBuf| -> bool { d.is_file() && d.file_name().unwrap_or_default() == "Cargo.toml" };
+
+    is_rs_file(&dir_entry) || is_cargo_toml(&dir_entry)
+}
+
+fn has_js_files(dir_entry: &PathBuf) -> bool {
+    let is_js_file =
+        |d: &PathBuf| -> bool { d.is_file() && d.extension().unwrap_or_default() == "js" };
+    let is_node_modules =
+        |d: &PathBuf| -> bool { d.is_dir() && d.file_name().unwrap_or_default() == "node_modules" };
+    let is_package_json = |d: &PathBuf| -> bool {
+        d.is_file() && d.file_name().unwrap_or_default() == "package.json"
+    };
+
+    is_js_file(&dir_entry) || is_node_modules(&dir_entry) || is_package_json(&dir_entry)
+}
+
+fn get_package_version(context: &Context) -> Option<String> {
+    let is_rs_project = context.dir_files.iter().any(has_rs_files);
+    if !is_rs_project {
+        let is_js_project = context.dir_files.iter().any(has_js_files);
+        if !is_js_project {
+            return None;
+        } else {
+            let mut file = File::open("package.json").unwrap();
+            let mut data = String::new();
+            file.read_to_string(&mut data).unwrap();
+
+            let json: serde_json::Value = serde_json::from_str(&data).unwrap();
+            let version = format_version(json["version"].to_string());
```

Comment:
What happens here if `package.json` doesn't have a version?

Reply:
I get a `vnull` ... which means that the version is `null` and then I appended a `v`

![Shot 2019-04-26 at 08 06 12](https://user-images.githubusercontent.com/30328854/56789617-375f7b00-67fa-11e9-9d0e-0d9a9d3e1b28.jpg)

Reply:
What would be the ideal way to handle this scenario?

If `package.json`  doesn't have a version, should we hide the section? If we did, that might be confusing for users, _why does my prompt not work?!_

Or should it say something like ` 📦 NO_VER` or `📦 vN/A` or...?

Reply:
@Snuggle @matchai I'd personally vote for `📦 NO_VER`. What do you guys think?

Reply:
In spaceship it has always been `📦⚠️`, which has lead to _a lot_ of confusion. I'd be fine with changing it here.

I'd personally prefer not to show the version section at all, to be honest. 🤔
`📦 NO_VER` is fine with me if that's what most people prefer, but I'm afraid it may just be noisy if people are intentionally omitting a version.

Reply:
> but I'm afraid it may just be noisy if people are intentionally omitting a version.

@matchai Exactly what I was thinking ... let's leave it out!